lib/mrsk/cli/healthcheck.rb

```diff
@@ -11,36 +15,33 @@ def perform
       begin
         execute *MRSK.healthcheck.run
 
-        target = "Health check against #{MRSK.config.healthcheck["path"]}"
         attempt = 1
-
         begin
-          status = capture_with_info(*MRSK.healthcheck.curl)
-
-          if status == "200"
-            info "#{target} succeeded with 200 OK!"
+          case output = capture_with_info(*MRSK.healthcheck.curl, raise_on_non_zero_exit: false)
+          when "200"
+            true
+          when "000"
+            raise HealthcheckError, "failed to connect"
+          when /\A\d+\z/
+            raise HealthcheckError, "failed with status #{output}"
           else
-            raise HealthcheckError, "#{target} failed with status #{status}"
+            raise HealthcheckError, "failed to run: #{output}"
           end
-        rescue SSHKit::Command::Failed
+        rescue HealthcheckError => error
           if attempt <= MAX_ATTEMPTS
-            info "#{target} failed to respond, retrying in #{attempt}s..."
+            info "#{error.message}, retrying in #{attempt}s..."
             sleep attempt
             attempt += 1
-
             retry
           else
             raise
           end
         end
-      rescue SSHKit::Command::Failed, HealthcheckError => e
-        error capture_with_info(*MRSK.healthcheck.logs)
 
-        if e.message =~ /curl/
-          raise SSHKit::Command::Failed, "#{target} failed to return 200 OK!"
-        else
-          raise
-        end
+        info "Health check succeeded with 200 OK!"
```

Comment:
Indicate success at the end, if nothing raised/failed, rather than within output status parsing.